servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -537,6 +503,8 @@ public void channelWritabilityChanged(ChannelHandlerContext ctx) {
             if (ctx.channel().isWritable()) {
                 connection.channelOutboundListener.channelWritable();
             } else if (connection.flushStrategyHolder.currentStrategy().shouldFlushOnUnwritable()) {
+                // TODO(scott): if we have a flush per write operation, shouldFlushOnUnwritable is more challenging.
```

Comment:
this is also related to the larger "state management on the server" discussion too https://github.com/apple/servicetalk/issues/981

Reply:
more investigation is required, so wanted to make a mental note. however if this is the only thing preventing shifting to a model of flush-on-request I would be willing to compromise on this (assuming we can just always flush in this scenario or otherwise avoid deadlock).